File: reference/src/security/env-secret-store.ts
/**
 * Environment variable-backed SecretStore.
 * Wraps process.env behind the SecretStore interface.
 * Secret names are converted from kebab-case to UPPER_SNAKE_CASE.
 */

import type { SecretStore } from './interfaces.js';

/**
 * Convert a kebab-case secret name to UPPER_SNAKE_CASE env var name.
 * e.g., "github-token" → "GITHUB_TOKEN"
 */
function toEnvVar(name: string): string {
  return name.replace(/-/g, '_').toUpperCase();
}

/**
 * Create a SecretStore backed by environment variables.
 *
 * @param env - The environment object to read from (defaults to process.env).
 */
export function createEnvSecretStore(
  env: Record<string, string | undefined> = process.env,
): SecretStore {
  return {
    get(name: string): string | undefined {
      return env[toEnvVar(name)];
    },

    getRequired(name: string): string {
      const envVar = toEnvVar(name);
      const value = env[envVar];
      if (!value) {
        throw new Error(`Secret "${name}" not found: environment variable ${envVar} is not set`);
      }
      return value;
    },
  };
}

File: sdk-python/src/ai_sdlc/security/env_secret_store.py
"""Environment variable-backed SecretStore."""

from __future__ import annotations

import os


def _to_env_var(name: str) -> str:
    return name.replace("-", "_").upper()


class _EnvSecretStore:
    def __init__(self, env: dict[str, str] | None = None) -> None:
        self._env = env

    def _get_env(self, key: str) -> str | None:
        if self._env is not None:
            return self._env.get(key)
        return os.environ.get(key)

    def get(self, name: str) -> str | None:
        return self._get_env(_to_env_var(name))

    def get_required(self, name: str) -> str:
        env_var = _to_env_var(name)
        value = self._get_env(env_var)
        if not value:
            msg = (
                f'Secret "{name}" not found: '
                f"environment variable {env_var} is not set"
            )
            raise ValueError(msg)
        return value


def create_env_secret_store(
    env: dict[str, str] | None = None,
) -> _EnvSecretStore:
    """Create a SecretStore backed by environment variables.

    Args:
        env: Optional dict to use instead of ``os.environ``.
    """
    return _EnvSecretStore(env)
